### DISCUSSION TITLE
NAS not listed in library, but mounted - SOLVED problem with Synology NFS share

### DISCUSSION THREAD

**Discussion by Sn4kY** [rern/rAudio#349] [Fixed Issues]
Hi
My NAS is correctly mounted :
`192.168.1.111:/volume1/Multimedia/MP3  7.0T  5.2T  1.9T  74% /mnt/MPD/NAS/NAS`
but rAudio's says 
```
Network storage not available.
To setup:
Settings > System - Storage - 
```
I don't understand why.
Also, no files inside have been indexed

Fresh install, with upgrade.

(No problem with previous installation with Re+3)

(thanks for you'r excellent job !)

**Comment by rern** [comment]
There's a bug in last update. 
Get today's update then remove > mount again.

**Reply by Sn4kY** [reply]
Hi.
Thanks for your reply, but, the problem still exists with 20210531 ; no NAS, even with a fresh install, RPi 3, connected with LAN cable

**Comment by Sn4kY** [comment]
Hi again.

I've straced the mpd process while manual mpc update command beeing processed, and got :
```
[pid 10018] readlink("/mnt/MPD/NAS/Sn4kY-NAS", 0x6c3612fc, 4096) = -1 EINVAL (Invalid argument)
[pid 10018] stat64("/mnt/MPD/NAS/Sn4kY-NAS", {st_mode=S_IFDIR|000, st_size=584, ...}) = 0
[pid 10018] openat(AT_FDCWD, "/mnt/MPD/NAS/Sn4kY-NAS", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_CLOEXEC|O_DIRECTORY) = -1 EACCES (Permission denied)
[pid 10018] write(2, "exception: Failed to open /mnt/MPD/NAS/Sn4kY-NAS: Permission denied\n", 68) = 68
[pid 10018] getdents64(18, 0x6ba08a68 /* 0 entries */, 32768) = 0
```
That's probably a clue

I've noticed a strange behaviour just after mpc update command :
```
rAudio:/mnt/MPD/NAS  #  ls -al
total 8
drwxr-xr-x 3 mpd audio 4096 Jun 27 11:31 .
drwxr-xr-x 5 mpd audio 4096 Apr 16 04:44 ..
d--------- 1 mpd audio  584 May 29 19:55 Sn4kY-NAS
```
and, after few minutes, it's back to normal again
```
rAudio:/mnt/MPD/NAS  #  ls -al
total 8
drwxr-xr-x 3 mpd audio 4096 Jun 27 11:31 .
drwxr-xr-x 5 mpd audio 4096 Apr 16 04:44 ..
drwxrwxrwx 1 mpd audio  584 May 29 19:55 Sn4kY-NAS
```

Maybe it's a problem with the NFS mount ; I remount my NAS with CIFS, and the Library is updating now, the folder right's did't change :
```
drwxr-xr-x 2 mpd mpd      0 Dec  6  2020 NAS
drwxrwxrwx 1 mpd audio  584 May 29 19:55 Sn4kY-NAS
```

I've noticed with `journalctl -xe` that when the NFS mount was added, an error with inotify was generated :
`Jun 27 12:19:44 rAudio taskset[4551]: exception: Failed to register /mnt/MPD/NAS/Sn4kY-NAS: inotify_add_watch() has failed: Permission denied`
but I don't know if this error is beacause of the directory's rights, or if it is the source of the problem (I.E : nfs supports inotify ?)

**Comment by Sn4kY** [comment]
problem solved, it's an permission issue on my Synology NAS side.

For the records, i've mapped all users to guests, and give to guest read access to the directory
